core/trino-spi/src/main/java/io/trino/spi/block/VariableWidthBlock.java

```diff
@@ -85,6 +86,23 @@ public VariableWidthBlock(int positionCount, Slice slice, int[] offsets, Optiona
         retainedSizeInBytes = INSTANCE_SIZE + slice.getRetainedSize() + sizeOf(valueIsNull) + sizeOf(offsets);
     }
 
+    /**
+     * Gets the raw {@link Slice} that keeps the actual data bytes.
+     */
+    public Slice getRawSlice()
```

Comment:
move these methods to `AbstractVariableWidthBlock`. It will be more generic then

Reply:
talked offline, AbstractVariableWidthBlock may not support one single Slice. I'm leaving this here only.